cmd/hamctl/command/promote.go

```diff
@@ -21,43 +18,31 @@ func NewPromote(client *httpinternal.Client, service *string) *cobra.Command {
 			})
 		},
 		RunE: func(c *cobra.Command, args []string) error {
-			committerName, committerEmail, err := git.CommitterDetails()
-			if err != nil {
-				return err
+			if fromEnvironment == "" {
+				switch toEnvironment {
+				case "dev":
+					fromEnvironment = "master"
+				case "staging":
+					fromEnvironment = "dev"
+				case "prod":
+					fromEnvironment = "staging"
+				}
 			}
-
-			var resp httpinternal.PromoteResponse
-			url, err := client.URL("promote")
+			artifactID, err := actions.ArtifactIDFromEnvironment(client, *service, namespace, fromEnvironment)
 			if err != nil {
 				return err
 			}
-			err = client.Do(http.MethodPost, url, httpinternal.PromoteRequest{
-				Service:        *service,
-				Environment:    environment,
-				Namespace:      namespace,
-				CommitterName:  committerName,
-				CommitterEmail: committerEmail,
-			}, &resp)
+
 			if err != nil {
 				return err
 			}
-			fmt.Printf("Promote service: %s\n", *service)
-			if resp.Status != "" {
-				fmt.Printf("%s\n", resp.Status)
-			} else {
-				fmt.Printf("[✓] Promotion of %s from %s to %s initialized\n", resp.Tag, resp.FromEnvironment, resp.ToEnvironment)
-			}
 
-			return nil
+			return actions.ReleaseArtifactID(client, *service, toEnvironment, artifactID, intent.NewPromoteEnvironment(fromEnvironment))
 		},
 	}
-	command.Flags().StringVarP(&environment, "env", "e", "", "Environment to promote to (required)")
-	completion.FlagAnnotation(command, "env", "__hamctl_get_environments")
-	// errors are skipped here as the only case they can occour are if thee flag
-	// does not exist on the command.
-	//nolint:errcheck
-	command.MarkFlagRequired("env")
-	command.Flags().StringVarP(&namespace, "namespace", "n", "", "Namespace the service is deployed to (defaults to env)")
-	completion.FlagAnnotation(command, "namespace", "__hamctl_get_namespaces")
+	command.Flags().StringVarP(&toEnvironment, "env", "", "", "Alias for '--to-env' (deprecated)")
```

Comment:
I strongly disagree with this change 😄 
I understand that it can be confusing whether it is the to or from environment and that knowledge has probably been tribal since the creation of the command.

I'd like to challenge whether this is generally misunderstood/hard to grasp and then take action when we know more. I also think that this is not directly affecting the goal of this PR relate to changing the flow? If that's the case can we move it into another branch and then discuss the change out of band with the rest of the changes?

Reply:
Reverting this, so this can be discussed elsewhere 👍